module/zfs/dmu.c

```diff
@@ -965,6 +965,9 @@ int
 dmu_read_by_dnode(dnode_t *dn, uint64_t offset, uint64_t size, void *buf,
     uint32_t flags)
 {
+	if (size == 0)
+		return (0);
```

Comment:
Why was this added?  I don't see why it should be needed here or in the write path.